py/surveyqa/summary.py

```diff
@@ -158,29 +157,26 @@ def get_summarytable(exposures):
 
     Returns a bokeh DataTable object.
     '''
-    night = exposures['NIGHT']
-    sorted_total = OrderedDict(Counter(night))
-    night_exps_total = np.array(sorted_total.values())
-
-    #list of counts of each program for each night
-    dct = []
-    for n in list(sorted_total.keys()):
-        keep = exposures['NIGHT'] == n
-        nights_selected = exposures[keep]
-        program_freq = Counter(nights_selected['PROGRAM'])
-        dct.append(program_freq)
-
-    #get the values out of the dictionaries above (in a specified order so we can splice later)
-    counts = []
-    for d in dct:
-        for key in ['BRIGHT', 'GRAY', 'DARK', 'CALIB']:
-            counts.append(d[key])
-
-    #lists of counts of each program for each night
-    brights = np.array([counts[i] for i in np.arange(0,len(counts), 4)])
-    grays = np.array([counts[i] for i in np.arange(1,len(counts), 4)])
-    darks = np.array([counts[i] for i in np.arange(2,len(counts), 4)])
-    calibs = np.array([counts[i] for i in np.arange(3,len(counts), 4)])
+    nights = np.unique(exposures['NIGHT'])
+
+    isbright = (exposures['PROGRAM'] == 'BRIGHT')
+    isgray = (exposures['PROGRAM'] == 'GRAY')
+    isdark = (exposures['PROGRAM'] == 'DARK')
+    iscalib = (exposures['PROGRAM'] == 'CALIB')
```

Comment:
Pre-filter which exposures are dark/gray/bright/calib in vectorized operations instead of loops with night filters and `Counter`.  These will be used later in a single loop over nights.